libraries/AP_GPS/AP_GPS_ERB.cpp

```diff
@@ -154,9 +154,9 @@ AP_GPS_ERB::_parse_gps(void)
     case MSG_POS:
         Debug("Message POS");
         _last_pos_time        = _buffer.pos.time;
-        state.location.lng    = (int32_t)(_buffer.pos.longitude * 1e7);
-        state.location.lat    = (int32_t)(_buffer.pos.latitude * 1e7);
-        state.location.alt    = (int32_t)(_buffer.pos.altitude_msl * 1e2);
+        state.location.lng    = (int32_t)(_buffer.pos.longitude * (double)1e7);
+        state.location.lat    = (int32_t)(_buffer.pos.latitude * (double)1e7);
+        state.location.alt    = (int32_t)(_buffer.pos.altitude_msl * (double)1e2);
```

Comment:
we can just use 100 here instead of (double)1e2. There are 3 more like this below.